gateway/modules/auth/match_test.go

```diff
@@ -481,11 +512,26 @@ func Test_matchHash(t *testing.T) {
 			args:    args{rule: &config.Rule{Rule: "hash", Fields: []string{"abc.password"}}, args: map[string]interface{}{"abc": map[string]interface{}{"password": "password"}}},
 			wantErr: true,
 		},
+		{
+			name: "rule clause - allow",
+			m:    &Module{project: "project"},
+			args: args{
+				rule: &config.Rule{Rule: "encrypt", Clause: &config.Rule{Rule: "allow"}},
+			},
+			want: &model.PostProcess{},
+		},
+		{
+			name: "rule clause - deny",
+			m:    &Module{project: "project"},
+			args: args{
+				rule: &config.Rule{Rule: "encrypt", Clause: &config.Rule{Rule: "deny"}},
+			},
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := matchHash(context.Background(), tt.args.rule, tt.args.args)
+			got, err := tt.m.matchHash(context.Background(), "project", tt.args.rule, tt.args.args, make(map[string]interface{}))
```

Comment:
We should check if the value was actually hashed or not

Reply:
👍